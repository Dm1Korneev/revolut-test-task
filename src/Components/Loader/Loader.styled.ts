
import styled, { keyframes } from 'styled-components';

const spinner = keyframes`
    0% {
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
`;

const spinnerSize = '5rem';

export const Container = styled.div`
    width: 100%;
    height: 100%;
    flex-grow: 1;
    text-align: center;
    z-index: 100;
    background-color: rgb(255,255,255);
    display: flex;
    justify-content: center;
    align-items: center;
`;

export const Spinner = styled.div`
    position: relative;
    width: ${spinnerSize};
    height: ${spinnerSize};
    margin-left: auto;
    margin-right: auto;

    div {
        box-sizing: border-box;
        display: block;
        position: absolute;
        width: ${spinnerSize};
        height: ${spinnerSize};
        border-radius: 50%;
        border-width: 8px;
        border-style: solid;
        border-color: rgb(235, 0, 141) transparent transparent transparent;
        animation: ${spinner} 1.3s cubic-bezier(0.49, 0.01, 0.56, 0.9) infinite;

        &:nth-child(1) {
            animation-delay: -0.45s;
        }

        &:nth-child(2) {
            animation-delay: -0.3s;
        }

        &:nth-child(3) {
            animation-delay: -0.15s;
        }
    }
`;
