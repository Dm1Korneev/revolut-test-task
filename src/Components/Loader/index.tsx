import React, { FC } from 'react';

import { Container, Spinner } from './Loader.styled';

const Loader: FC = (): JSX.Element => (
  <Container>
    <Spinner>
      <div />
      <div />
      <div />
      <div />
    </Spinner>
  </Container>
);

export default Loader;
