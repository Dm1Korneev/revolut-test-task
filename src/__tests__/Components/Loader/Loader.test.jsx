import React from 'react';
import { render } from '@testing-library/react';
import Loader from 'Components/Loader/';

describe('Loader component', () => {
  test('renders the component', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
