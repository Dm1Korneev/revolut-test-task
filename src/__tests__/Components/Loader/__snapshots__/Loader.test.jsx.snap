// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa iFHshV"
  >
    <div
      class="sc-bwzfXH bsnhCc"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
